src/base/exceptions.h

```diff
@@ -28,12 +28,30 @@
 
 #pragma once
 
-#include <stdexcept>
+#include <exception>
 #include <QString>
 
-class RuntimeError : public std::runtime_error
+class Exception : public std::exception
 {
 public:
-    explicit RuntimeError(const QString &message = {});
-    QString message() const;
+    explicit Exception(const QString &message = {});
+
+    [[nodiscard]] QString message() const noexcept;
```

Comment:
Two questions here:
1. Is it a good idea to use these suggestive specifiers (nodiscard, noexcept) ?
   Currently I think they are too verbose (add noise) and I haven't found a valid use case for them.
2. I doubt this function is `noexcept`. If I understand it correctly, it will return a new `QString` object and hence is subject to allocate memory (without the implicit sharing part) and it might fail under extreme conditions and throw exception such as `bad_alloc`.

Reply:
> Is it a good idea to use these suggestive specifiers (nodiscard, noexcept) ?

IIRC, `[[nodiscard]]` is just hint to the compiler to prevent meaningless use of functions that have no side effect (e.g. getters).
`noexcept` has a more advanced meaning, declaring the intention to provide some method as exception-free. Of course, nothing prevents you from providing such methods without any `noexcept` specifier. But it makes it easier to create a more complex hierarchy. For example, here the method is intended to be used in an exception handler, so it must be exception-free, otherwise it cannot be reliably used there. Again, I can omit this specifier for this method, but I won't be able to do it for `what()`, since it's originally declared as `noexcept`, and some compilers don't allow overriding it without this specifier.
> Currently I think they are too verbose (add noise) and I haven't found a valid use case for them.

Maybe it looks too verbose, because we are not used to seeing such code. I don't feel that we should reject it because of its verbosity, but neither do I insist on its widespread use.
Of course I can omit them everywhere they aren't strictly required.

> I doubt this function is `noexcept`. If I understand it correctly, it will return a new `QString` object and hence is subject to allocate memory (without the implicit sharing part) and it might fail under extreme conditions and throw exception such as `bad_alloc`.

There is `QString(const QString &other) noexcept` so we can rely on it.
Really it performs no memory allocation due to its "implicit sharing".